oio/directory/indexer.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (C) 2018 OpenIO SAS, as part of OpenIO SDS
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from datetime import datetime
+from oio.blob.utils import check_volume_for_service_type
+from oio.common import exceptions as exc
+from oio.common.constants import STRLEN_REFERENCEID
+from oio.common.daemon import Daemon
+from oio.common.easy_value import int_value
+from oio.common.green import ratelimit, ContextPool
+from oio.common.http_urllib3 import get_pool_manager
+from oio.common.logger import get_logger
+from oio.common.utils import paths_gen
+from oio.directory.client import DirectoryClient
+from oio.rdir.client import RdirClient
+
+import time
```

Comment:
We have had troubles with `time.time()` and `time.sleep()` recently. It's better to import them from `oio.common.green`.